File: tests/semantic_gold_fixtures/rust/clean_complex.rs
pub fn clean_parse_tokens(text: &str) -> Vec<String> {
    let mut tokens = Vec::new();
    let mut current = String::new();
    let mut quoted = false;
    let mut escaped = false;
    for ch in text.chars() {
        if escaped {
            current.push(ch);
            escaped = false;
        } else if ch == '\\' {
            current.push(ch);
            escaped = true;
        } else if ch == '"' {
            current.push(ch);
            quoted = !quoted;
        } else if ch.is_whitespace() && !quoted {
            if !current.is_empty() {
                tokens.push(std::mem::take(&mut current));
            }
        } else {
            current.push(ch);
        }
    }
    if !current.is_empty() {
        tokens.push(current);
    }
    tokens
}

pub fn clean_validate_contract(result: &serde_json::Value) -> &str {
    let tier = result.get("tier").and_then(|value| value.as_str()).unwrap_or("");
    let smelly = result.get("smelly").and_then(|value| value.as_bool()).unwrap_or(false);
    if smelly != (tier != "clean") {
        panic!("tier and smelly disagree");
    }
    tier
}

pub fn kinda_forward_payload(payload: Payload) -> Payload {
    let forwarded = payload;
    forwarded
}

File: tests/semantic_gold_fixtures/go/clean_complex.go
package gold

type Payload struct{}

var (
	_ = cleanParseTokens("")
	_ = cleanValidateContract(map[string]any{"tier": "clean", "smelly": false})
	_ = kindaForwardPayload(Payload{})
)

func cleanParseTokens(text string) []string {
	tokens := []string{}
	current := []rune{}
	quoted := false
	escaped := false
	for _, ch := range text {
		if escaped {
			current = append(current, ch)
			escaped = false
		} else if ch == '\\' {
			current = append(current, ch)
			escaped = true
		} else if ch == '"' {
			current = append(current, ch)
			quoted = !quoted
		} else if ch == ' ' && !quoted {
			if len(current) > 0 {
				tokens = append(tokens, string(current))
				current = nil
			}
		} else {
			current = append(current, ch)
		}
	}
	if len(current) > 0 {
		tokens = append(tokens, string(current))
	}
	return tokens
}

func cleanValidateContract(result map[string]any) string {
	tier, _ := result["tier"].(string)
	smelly, _ := result["smelly"].(bool)
	if smelly != (tier != "clean") {
		panic("tier and smelly disagree")
	}
	return tier
}

func kindaForwardPayload(payload Payload) Payload {
	forwarded := payload
	return forwarded
}
